src/Extension/StatusRuntime.php

```diff
@@ -35,9 +36,32 @@ public function addStatusService(StatusClassRendererInterface $renderer): void
     }
 
     /**
-     * @param mixed $statusType
+     * @param object|string $object     Object for StatusClassRenderer or string for FlashManager
+     * @param string|null   $statusType Object status type or Sonata flash message type
+     * @param string        $default    Default status class
+     * @param mixed         $statusType
      */
-    public function statusClass(object $object, $statusType = null, string $default = ''): string
+    public function statusClass($object, $statusType = null, string $default = ''): string
+    {
+        if (\is_object($object)) {
+            return $this->statusClassForStatusClassRenderer($object, $statusType, $default);
+        }
+
+        if (\is_string($object)) {
+            return $this->statusClassForFlashManager($object, $statusType, $default);
+        }
+
+        @trigger_error(sprintf(
+            'Argument 1 passed to "%s()" must be an instance of "%s" or a string, %s given.',
+            __METHOD__,
+            StatusClassRendererInterface::class,
+            \is_object($object) ? 'instance of '.\get_class($object) : \gettype($object)
+        ));
```

Comment:
```suggestion
            'Passing other type than object or string as argument 1 for "%s()" is deprecated since sonata-project/twig-extensions 1.x'
            .' and will throw an exception in version 2.0.' 
            __METHOD__
        ), E_USER_DEPRECATED);
```

Reply:
I remove `object` type hint to keep BC. It should be revert in 2.0. This message is little confusing for it. IMHO current message is OK. I will add in next `Improve FlashManager` another `error_triger` to tell people to avoid using `string` and use FlashManager directly instead.

Reply:
I'm dizzy. Is `$object` intended to be an instance of `StatusClassRendererInterface`? If not, the current deprecation message is wrong.

Reply:
Oh, crap. Should be object or string. Can you recheck it?

Reply:
IMO, the message you're using currently will be good for the exception, but it is not the best choice for a deprecation, where the version which deprecates the behavior must be explicitly exposed.
Even, please note that you're omitting the 2nd argument for `trigger_error()`, which for a deprecation should be `E_USER_DEPRECATED`.
I can't propose a better alternative than I've left at the beginning of this thread.

Reply:
i update it. This message can be modify in next PR, when I update documentation about FlashManager.